app/src/main/kotlin/no/nav/tiltakspenger/saksbehandling/saksbehandling/service/behandling/IverksettBehandlingService.kt

```diff
@@ -127,38 +126,46 @@ class IverksettBehandlingService(
         vedtak: Rammevedtak,
         sakStatistikk: StatistikkSakDTO,
         stønadStatistikk: StatistikkStønadDTO,
-    ) {
-        val førsteMeldeperiode = this.opprettFørsteMeldeperiode()
+    ): Sak {
+        val (oppdatertSak, meldeperioder) = this.genererMeldeperioder()
+        // Denne har vi behov for å gjøre ved påfølgende førstegangsbehandligner (altså ikke den første)
+        val (oppdaterteMeldekortbehandlinger, oppdaterteMeldekort) =
+            this.meldekortBehandlinger.oppdaterMedNyeKjeder(oppdatertSak.meldeperiodeKjeder, tiltakstypeperioder)
 
         // journalføring og dokumentdistribusjon skjer i egen jobb
         // Dersom denne endres til søknadsbehandling og vi kan ha mer enn 1 for en sak og den kan overlappe den eksistrende saksperioden, må den legge til nye versjoner av meldeperiodene her.
         sessionFactory.withTransactionContext { tx ->
             behandlingRepo.lagre(vedtak.behandling, tx)
+            sakService.oppdaterSisteDagSomGirRett(oppdatertSak.id, oppdatertSak.sisteDagSomGirRett, tx)
             rammevedtakRepo.lagre(vedtak, tx)
             statistikkSakRepo.lagre(sakStatistikk, tx)
             statistikkStønadRepo.lagre(stønadStatistikk, tx)
-            meldeperiodeRepo.lagre(førsteMeldeperiode, tx)
+            oppdaterteMeldekort.forEach { meldekortBehandlingRepo.oppdater(it, tx) }
+            meldeperioder.forEach {
```

Comment:
Trodde vi hadde en metode for å sende hele lista? Men husk at det er viktig og sende/bruke tx.

Reply:
Yep, lagde den etter 😁